iib/workers/tasks/build.py

```diff
@@ -453,6 +443,27 @@ def handle_add_request(bundles, binary_image, request_id, from_index=None, add_a
     if from_index:
         _verify_index_image(prebuild_info['from_index_resolved'], from_index)
 
+    legacy_support_packages = get_legacy_support_packages(bundles)
+    if legacy_support_packages and not all([cnr_token, organization]):
+        packages_str = ', '.join(legacy_support_packages)
+        raise IIBError(
+            f'Legacy support is required for {packages_str};'
+            f' Both cnr_token and organization should be non-empty strings'
+        )
+
+    if legacy_support_packages:
+        conf = get_worker_config()
+        rebuilt_index_image = conf['iib_image_push_template'].format(
+            registry=conf['iib_registry'], request_id=request_id
+        )
+        opm_index_export(
+            legacy_support_packages, request_id, rebuilt_index_image, cnr_token, organization
+        )
+
+        if from_index:
+            # Re-verify index image after exporting it for legacy app registry
```

Comment:
Ah, nice catch! I should call the `_push_package_manifest` method after re-checking the from_index. 
The reason I do this is to check if the index image has changed since the IIB build started. This is because we cannot use the resolved image in the opm command. So after exporting the package, if the source index image has changed, IIB will fail the request (doesn't maater if the build and export was successful). Does that make sense?

Reply:
It does. Can you make sure to add this explanation to the code?